mlir/include/mlir/Dialect/Vector/IR/VectorOps.td

```diff
@@ -444,6 +444,8 @@ def Vector_ShuffleOp :
       mask values must be within range, viz. given two k-D operands v1 and v2
       above, all mask values are in the range [0,s_1+t_1)
 
+    Note, at the moment only fixed-width vectors are supported.
```

Comment:
nit: I'd drop "at the moment" as it suggests support is missing but as you mention it seems unlikely this will change given we have dedicated ops like splat / broadcast and interleave.